README.md

```diff
@@ -3,103 +3,124 @@
 
 ## Writing a CAN spec
 The CAN spec should be a YAML file with the following structure:
 ```yaml
 name: MYwhatever # Name of your CAN spec
+units: # Any custom units you need to define
+  - newtonmeter = N * m = J = Nm
 architectures: # Allowed architectures for boards
   - arch0
   - arch1
-boards: # All boards on the car, along with CAN messages they publish and subscribe to
+computer: # All boards on the car, along with CAN messages they publish and subscribe to
   board1:
-    arch: arch1
-    publish:
-      can_bus0: # All messages on the bus "can_bus0" board1 publishes
-        - Msg0
-      can_bus1: # All messages on the bus "can_bus1" board1 publishes
-        - Msg1
-    subscribe: # All messages on : bus "can_bus0" board1 subscribes to
-      can_bus0:
-        - Msg2
-  # Note that the next board has no architecture specified
-  # (Good for boards we don't program ourselves)
-  board2:
-    publish:
-      can_bus0:
-        - Msg2
-    subscribe:
-      can_bus0:
-        - Msg0
-buses: # List of CAN buses on the car
-  can0: # All messages on can0
-    baudrate: 500000
-    extended: false
-    messages:
-      Msg0:
-        id: 0x001 # Note that lower CAN IDs have higher priority
-        period: 50ms # optional, only add it if you will use it
-        segments:
-          segment0:
-            slice: 0 + 32  # [start index] + [length]
-            type: int32 big  # type and endianness
-            unit: V  # optional
-          segment1:
-            slice: 32 + 32
-            type: int32 big
-            unit: V
-      Msg2:
-        can_id: 0x002
-        segments:
-          segment0:
-            slice: 0 + 2
-            type: enum
-            enumerations: # Make sure to add this for segments of type enum
-              ENUM_A: 0x0
-              ENUM_B: 0x1
-              ENUM_C: 0x2
-              ENUM_D: 0x3
-  can1:
-    baudrate: 1000000
-    extended: true
-    messages:
-      Msg0:
-        id: 0x001 # Note that lower CAN IDs have higher priority
-        segments:
-          segment0:
-            slice: 0 + 32  # [start index] + [length]
-            type: int32 big  # type and endianness
-            unit: V  # optional
-          segment1:
-            slice: 32 + 32
-            type: int32 big
-            unit: V
+    architecture: arch1
+    participation:
+      can:
+        mapping:
+          symbolic_bus_0: hardware_bus_2
+          symbolic_bus_1: hardware_bus_3
+        publish:
+          symbolic_bus_0:
+            - Msg0
+          symbolic_bus_1:
+            - Msg0
+        subscribe:
+          symbolic_bus_0:
+            - Msg2
+protocol:
+  can:
+    bus: # List of CAN buses on the car
+      symbolic_bus_0: # All messages on can0
```

Comment:
Let's not encourage underscores in the examples because they're used as a delimiter.